reactioncommerce/api-service-account
# Allow inviteShopMember to accept multiple groups

Resolves #9 
Impact: **breaking**
Type: **feature**

**Test with [api-plugin-email-templates#7](https://github.com/reactioncommerce/api-plugin-email-templates/pull/7)**

## Issue
The `inviteShopMember` mutation allows a `groupId` to be specified, but it would be even better to allow multiple groups with a `groupIds` array.

## Solution
Allow `inviteShopMember` to take a `groupIds` array instead of a `groupId` string.

## Breaking changes
- The `groupId` variable won't work anymore for `inviteShopMember`. To invite a new user to a single account, pass a single-item `groupIds` array to the mutation.
- The `inviteNewShopMember` template needs to be at its latest version, as the `groupName` variable was updated to `groupNames`.

## Testing
1. Call `inviteShopMember` with one group in the `groupIds` array.
2. Receive an e-mail referring to this group's name in a singular form.
3. Verify that, for this invite's object in the `AccountInvites` collection, the `groupIds` field has the correct value.
4. Sign up as the new account. Verify that the account is part of the expected group.
5. Call `inviteShopMember` with multiple groups in the `groupIds` array.
6. Receive an e-mail referring to the group names in a plural form.
7. Verify that, for this invite's object in the `AccountInvites` collection, the `groupIds` field has the correct value.
8. Sign up as the new account. Verify that the account is part of the expected groups.
